execute/tokendata/cctp/v2/deposit_hash.go

```diff
@@ -0,0 +1,188 @@
+// Package v2
+// Implements depositHash calculation for CCTPv2 Messages, which is needed to match CCTPv2
+// messages to SourceTokenDataPayloadV2 instances (which have a DepositHash field).
+package v2
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// depositHashABIArguments is the cached ABI arguments structure for deposit hash encoding.
+var depositHashABIArguments = initDepositHashABIArguments()
+
+// CalculateDepositHash calculates the depositHash from CCTPv2 message fields.
+// This must match Solidity's _calculateDepositHash in USDCSourcePoolDataCodec.sol.
+// The hash is: keccak256(abi.encode(sourceDomain, amount, destinationDomain, mintRecipient,
+// burnToken, destinationCaller, maxFee, minFinalityThreshold))
+//
+// This function is critical for matching CCIP token transfers with Circle's attestations.
+//
+// Parameters from msg are parsed and validated, then ABI-encoded in the exact order required
+// by the Solidity implementation. Any mismatch in encoding will result in hash mismatches and
+// failed attestation matching.
+func CalculateDepositHash(msg CCTPv2DecodedMessage) ([32]byte, error) {
+	var zero [32]byte
+
+	// Parse all parameters from the message
+	params, err := parseDepositHashParams(msg)
+	if err != nil {
+		return zero, err
+	}
+
+	// Use the cached ABI arguments structure
+	arguments := depositHashABIArguments
+
+	// Pack the arguments using ABI encoding
+	encoded, err := arguments.Pack(
+		params.sourceDomain,
+		params.amount,
+		params.destinationDomain,
+		params.mintRecipient,
+		params.burnToken,
+		params.destinationCaller,
+		params.maxFee,
+		params.minFinalityThreshold,
+	)
+	if err != nil {
+		return zero, fmt.Errorf("failed to ABI encode deposit hash parameters: %w", err)
+	}
+
+	// Calculate Keccak256 hash
+	hash := crypto.Keccak256Hash(encoded)
+	return hash, nil
+}
+
+// depositHashParams contains all parsed parameters needed for depositHash calculation.
+type depositHashParams struct {
+	sourceDomain         uint32
+	destinationDomain    uint32
+	amount               *big.Int
+	maxFee               *big.Int
+	minFinalityThreshold uint32
+	mintRecipient        [32]byte
+	burnToken            [32]byte
+	destinationCaller    [32]byte
+}
+
+// parseDepositHashParams extracts and validates all parameters from a CCTPv2DecodedMessage.
+// This function handles all string parsing and hex decoding, returning a structured
+// set of parameters ready for ABI encoding.
+func parseDepositHashParams(msg CCTPv2DecodedMessage) (depositHashParams, error) {
+	var params depositHashParams
+	var err error
+
+	// Parse sourceDomain
+	sourceDomainParsed, err := strconv.ParseUint(msg.SourceDomain, 10, 32)
```

Comment:
nit, but I see multiple parse calls like this. I think it could be slightly improved by extracting this logic into a separate function

```go
func parseUint32Param(param string, paramName string) (uint32, error) {
	parsed, err := strconv.ParseUint(param, 10, 32)
	if err != nil {
		return 0, fmt.Errorf("failed to parse %s: %w", paramName, err)
	}
	return uint32(parsed), nil
}

// and then call in `parseDepositHashParams`
	params.sourceDomain, err = parseUint32Param(msg.SourceDomain, "sourceDomain")
	if err != nil {
		return params, err
	}

	params.destinationDomain, err = parseUint32Param(msg.DestinationDomain, "destDomain")
	if err != nil {
		return params, err
	}

....
```

Reply:
I don't think this reduces complexity/LoC enough to be warranted imo

Reply:
Every parsing call has these magic numbers: 10 and 32. Having that in a single place reduces complexity, because if a change is required, you need to apply it only to a single place.